v0.7/medical_imaging/3d-unet/preprocess.py

```diff
@@ -0,0 +1,124 @@
+# coding=utf-8
+# Copyright (c) 2020 NVIDIA CORPORATION. All rights reserved.
+# Copyright 2020 Division of Medical Image Computing, German Cancer Research Center (DKFZ), Heidelberg, Germany
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import numpy
+import os
+import pickle
+import sys
+import torch
+
+sys.path.insert(0, os.path.join(os.getcwd(), "nnUnet"))
+
+from batchgenerators.augmentations.utils import pad_nd_image
+from batchgenerators.utilities.file_and_folder_operations import subfiles
+from nnunet.training.model_restore import load_model_and_checkpoint_files
+from nnunet.inference.predict import preprocess_multithreaded
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--model_dir", default="build/result/nnUNet/3d_fullres/Task043_BraTS2019/nnUNetTrainerV2__nnUNetPlansv2.mlperf.1",
+        help="Path to the directory containing plans.pkl")
+    parser.add_argument("--raw_data_dir", default="build/raw_data/nnUNet_raw_data/Task043_BraTS2019/imagesTr",
+        help="Path to the directory containing raw nii.gz files")
+    parser.add_argument("--preprocessed_data_dir", default="build/preprocessed_data", help="Path to the directory containing preprocessed data")
+    parser.add_argument("--validation_fold_file", default="fold4_validation.npy", help="Path to the npy file storing all the sample names for the validation fold")
+    parser.add_argument("--num_threads_preprocessing", type=int, default=12, help="Number of threads to run the preprocessing with")
+    args = parser.parse_args()
+    return args
+
+def preprocess_MLPerf(model, checkpoint_name, folds, fp16, list_of_lists, output_filenames, preprocessing_folder, num_threads_preprocessing):
+    assert len(list_of_lists) == len(output_filenames)
+
+    print("emptying cuda cache")
+    torch.cuda.empty_cache()
```

Comment:
Can you add a CUDA check?

if torch.cuda.is_available():
   torch.cuda.empty_cache()

Reply:
This way is will work on CPU

Reply:
I think I will just remove it. Not sure why we need to clean cache